lib/tesla_api/client.rb

```diff
@@ -53,7 +54,24 @@ def login!(password)
     end
 
     def vehicles
-      self.class.get("/vehicles")["response"].map { |v| Vehicle.new(self.class, email, v["id"], v) }
+      self.get("/vehicles")["response"].map { |v| Vehicle.new(self, email, v["id"], v) }
+    end
+
+    def self.base_uri
+      "https://owner-api.teslamotors.com/api/1"
+    end
+
+    def get( url )
+      HTTParty.get( self.class.base_uri + url, {headers: @headers, format: :json} )
```

Comment:
This should just be:
```ruby
def get(uri)
  self.class.get(uri, headers: headers)
end
```
Note that you have an `attr_reader` so you don't need to access the instance variable directly.